lib/retex.ex

```diff
@@ -11,15 +22,19 @@ defmodule Retex do
             bindings: %{},
             pending_activation: []
 
+  @spec root_vertex :: Retex.Root.t()
   def root_vertex(), do: Retex.Root.new()
 
+  @spec new :: Retex.t()
   def new() do
     %{graph: graph} = %Retex{}
     graph = Graph.add_vertex(graph, Retex.Root.new())
     %Retex{graph: graph}
   end
 
+  @spec add_wme(Retex.t(), Retex.Wme.t()) :: Retex.t()
   def add_wme(%Retex{} = network, %Retex.Wme{} = wme) do
+    wme = Map.put(wme, :timestamp, :os.system_time(:seconds))
```

Comment:
what about using put_new here so that if the value is already given it doesn't get overwritten? What do you think?

Reply:
the timestamp is only set here during addition, so i would leave it, and eventually override to ensure consistent order.